src/registry-client/src/index.ts

```diff
@@ -91,19 +102,41 @@ const agentOptions: Agent.Options = {
 const xdg = new XDG('vlt')
 
 export class RegistryClient {
-  agent: Agent
+  agent: RetryAgent
   cache: Cache
 
-  constructor({
-    cache = xdg.cache('registry-client'),
-  }: RegistryClientOptions) {
+  constructor(options: RegistryClientOptions) {
+    const {
+      cache = xdg.cache('registry-client'),
+      'fetch-retry-factor': timeoutFactor = 2,
+      'fetch-retry-mintimeout': minTimeout = 0,
+      'fetch-retry-maxtimeout': maxTimeout = 30_000,
+      'fetch-retries': maxRetries = 3,
+    } = options
```

Comment:
Generally, I'm not a fan of doubling up on these defaults in both the CLI and the workspace. It's already a pattern we use everywhere so I'm ok with adding another instance here.

I'd rather have our workspaces be less ergonomic and rely on types to force setting the options. I think this pattern is less likely to lead to bugs in the future if CLI defaults get changed or removed.

Again, not blocking this PR but curious to pros to this pattern that I might not be seeing.

Reply:
Good call. Updated, PTAL!